lib/iris/analysis/_interpolate_private.py

```diff
@@ -193,10 +193,13 @@ def nearest_neighbour_indices(cube, sample_points):
     return tuple(indices)
 
 
-def _nearest_neighbour_indices_ndcoords(cube, sample_point, cache=None):
+def _nearest_neighbour_indices_ndcoords(cube, sample_points, cache=None):
```

Comment:
this function is now quite clever.  Would it be plausible to have a unit test which threw some known and soluble problems at this function at asserted that the expected answers were produced?